jl4-core/src/L4/TypeCheck/Types.hs

```diff
@@ -292,20 +294,86 @@ resolvedType n = do
         KnownType kind _resolved _tyDecl -> setAnnResolvedKindOfResolved kind n
         KnownTerm ty _term -> setAnnResolvedTypeOfResolved ty n
         KnownTypeVariable -> setAnnResolvedKindOfResolved 0 n
+        KnownSection _ -> n -- TODO: this is probably not what we want, maybe some internal error?
```

Comment:
It looks strange to me that we couldn't be actually doing anything meaningful here?

Reply:
well a section doesn't have a type and this function is only useful for setting the type of that known thing

Reply:
Sections do have types!

Reply:
I have had a another look and I think this is for hovering type information. While I understand what you said wrt the types of sections, I don't think we want to show them in the user interface, right? So I guess we cannot do anything useful here, still, can we?